src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,290 @@
+      ringParticles.splice(0, p.floor((ringParticles.length - MAX_RING_PARTICLES * 0.8) * EASING));
+    }
+
+    // Loop through the array and show each particle
+    for (let i = 0; i < ringParticles.length; i++) {
+      ringParticles[i].update();
+      if (ringParticles[i].delete()) {
+        ringParticles.splice(i, 1);
+      } else {
+        ringParticles[i].show();
+      }
+    }
+  };
+  // Spawn ringParticles in ring with centre mouse x and y coords
+  p.mouseClicked = () => {
+    for (let i = 0; i < RING_COVERAGE; i += 4) {
+      const c = new ringParticle(p.mouseX, p.mouseY, i);
+      ringParticles.push(c);
+    }
+  };
+
+  class Firework {
+    explodeSize: number;
+    firework: Particle;
+    exploded: boolean;
+    particles: Array<Particle>;
+
+    constructor() {
+      this.explodeSize = p.random(0, EXPLODE_RANGE);
+      this.firework = new Particle(p.random(p.width), p.height, true);
+      this.exploded = false;
+      this.particles = [];
+    }
+
+    update() {
+      if (!this.exploded) {
+        this.firework.applyForce(gravity);
+        this.firework.update();
+
+        if (this.firework.vel.y >= 0) {
+          this.exploded = true;
+          this.explode();
+        }
+      }
+
+      for (let i = this.particles.length - 1; i >= 0; i--) {
+        this.particles[i].applyForce(gravity);
+        this.particles[i].update();
+
+        if (this.particles[i].done()) {
+          this.particles.splice(i, 1);
+        }
+      }
+    }
+
+    done() {
+      if (this.exploded && this.particles.length === 0) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    show() {
+      if (!this.exploded) {
+        this.firework.show();
+      }
+
+      for (let i = 0; i < this.particles.length; i++) {
+        this.particles[i].show();
+      }
+    }
+
+    explode() {
+      for (let i = 0; i < this.explodeSize * EXPLODE_OFFSET; i++) {
+        const particle = new Particle(this.firework.pos.x, this.firework.pos.y, false);
+        this.particles.push(particle);
+      }
+    }
+  }
+
+  class Particle {
+    pos: p5.Vector;
+    firework: boolean;
+    lifespan: number;
+    acc: p5.Vector;
+    ran: number;
+    vel: p5.Vector;
+
+    constructor(x: number, y: number, firework: boolean) {
+      this.pos = p.createVector(x, y);
+      this.firework = firework;
+      this.lifespan = FIREWORK_LIFESPAN;
+      this.acc = p.createVector(0, 0);
+      this.ran = p.random(0, 3);
+      if (this.firework) {
+        this.vel = p.createVector(0, p.random(-12, -8));
+      } else {
+        this.vel = p5.Vector.random2D();
+        this.vel.mult(p.random(2, 10));
+      }
+    }
+
+    applyForce(force: p5.Vector) {
+      this.acc.add(force);
+    }
+
+    update() {
+      if (!this.firework) {
+        this.vel.mult(0.9);
+        this.lifespan -= 4;
+      }
+
+      this.vel.add(this.acc);
+      this.pos.add(this.vel);
+      this.acc.mult(0);
```

Comment:
if we are setting `this.acc` to zero every time.. but also applying a force.. why not just do the assignment to the force and not have to do the multiplication?